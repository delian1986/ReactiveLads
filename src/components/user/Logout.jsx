import PropTypes from "prop-types";
import { Button } from "../common/Button";

export const Logout = ({ logout }) => (
  <Button
    handleClick={logout}
    type="button"
    className="btn btn-link nav-link"
    label="Logout"
  />
);

Logout.propTypes = {
  logout: PropTypes.func
};
