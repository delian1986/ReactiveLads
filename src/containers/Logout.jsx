import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { logoutAsync } from "../store/actions/auth";
import { Logout as LogoutView } from "../components/user/Logout";

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(
    {
      logout: () => logoutAsync()
    },
    dispatch
  );
};

export default connect(null, mapDispatchToProps)(LogoutView);
