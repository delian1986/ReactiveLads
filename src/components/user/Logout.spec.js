import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import Logout from "../../containers/Logout";

const mockStore = configureStore([]);

describe("tests <Logout/> component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({});
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <Logout />
      </Provider>
    );
  });
  describe("Test rendering component", () => {
    it("should render with given state from Redux store", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe("Actions", () => {
    it("should dispatch an action on button click", () => {
      renderer.act(() => {
        component.root.findByType("button").props.onClick();
      });

      expect(store.dispatch).toHaveBeenCalledTimes(1);
    });
  });
});
