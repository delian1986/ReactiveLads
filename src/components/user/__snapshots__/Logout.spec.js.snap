// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <Logout/> component Test rendering component should render with given state from Redux store 1`] = `
<button
  className="btn btn-link nav-link"
  onClick={[Function]}
  type="button"
>
  Logout
</button>
`;
